iox_data_generator/src/specification.rs

```
@@ -76,25 +88,10 @@ impl DataSpec {
         let mut bucket_agents = Vec::with_capacity(buckets.len());
 
         let mut start = 0;
-        for w in &self.bucket_writers {
-            if start >= buckets.len() {
-                warn!(
-                    "Bucket_writers percentages > 1.0. Writer {:?} and later skipped.",
-                    w
-                );
-                break;
-            }
-
-            // validate the agents actually exist in the spec
-            for a in &w.agents {
-                let _ = self.agent_by_name(&a.name).unwrap_or_else(|| {
-                    panic!(
-                        "agent {} referenced in bucket writers, but isn't in the spec",
-                        a.name
-                    )
-                });
-            }
+        let use_ratio =
```

Would it make sense to do `use_ratio = <if any bucker writer has ratio set>` rather than just checking the first one?

When written this way it seems like if some writer other than the first has a `ratio` specified, it will be silently ignored in favor of `regex`